opm/autodiff/AutoDiffHelpers.hpp

```diff
@@ -127,11 +232,15 @@ struct HelperOps
             const IFIndex nif = h.internal_faces.size();
             typename ADFaceCellTraits<Grid>::Type
                 face_cells = faceCellsToEigen(g);
-            assert(nif == ifaceflux.size());
+
+            // num connections may possibly include NNCs
+            size_t num_nnc = h.nnc_trans.size();
+            size_t num_connections= nif + num_nnc;
+            assert(num_connections == ifaceflux.size());
```

Comment:
This line triggers a signed/unsigned comparison warning, as well as line 257. Eigen sizes are ints, so using `int` for num_connections and num_nnc is the simplest way to resolve this. Also use spaces around `=` in line 238.